Sources/Apodini/Semantic Model Builder/REST/RESTEndpointHandler.swift

```diff
@@ -2,79 +2,42 @@
 // Created by Andi on 30.12.20.
 //
 
-import Foundation
 @_implementationOnly import Vapor
-import protocol FluentKit.Database
 
-extension Vapor.Request: ExporterRequest, WithEventLoop {}
-
-struct ResponseContainer: Encodable, ResponseEncodable {
-    typealias Links = [String: String]
-    var data: AnyEncodable?
-    var links: Links
-
-    enum CodingKeys: String, CodingKey {
-        case data = "data"
-        case links = "_links"
-    }
-
-    init(_ data: AnyEncodable?, links: [String: String]) {
-        self.data = data
-        self.links = links
-    }
-
-    func encodeResponse(for request: Vapor.Request) -> EventLoopFuture<Vapor.Response> {
-        let jsonEncoder = JSONEncoder()
-        jsonEncoder.outputFormatting = [.withoutEscapingSlashes, .prettyPrinted]
-
-        let response = Vapor.Response()
-        do {
-            if data != nil {
-                try response.content.encode(self, using: jsonEncoder)
-            }
-        } catch {
-            return request.eventLoop.makeFailedFuture(error)
-        }
-        return request.eventLoop.makeSucceededFuture(response)
-    }
-}
-
-class RESTEndpointHandler<H: Handler> {
-    let endpoint: Endpoint<H>
-    var contextCreator: () -> AnyConnectionContext<RESTInterfaceExporter>
+struct RESTEndpointHandler {
+    var contextCreator: () -> ConnectionContext<RESTInterfaceExporter>
     let configuration: RESTConfiguration
 
-    init(
-        for endpoint: Endpoint<H>,
-        using contextCreator: @escaping () -> AnyConnectionContext<RESTInterfaceExporter>,
-        configuration: RESTConfiguration) {
-        self.endpoint = endpoint
-        self.contextCreator = contextCreator
+    init(configuration: RESTConfiguration, using contextCreator: @escaping () -> ConnectionContext<RESTInterfaceExporter>) {
         self.configuration = configuration
+        self.contextCreator = contextCreator
     }
 
     func register(at routesBuilder: Vapor.RoutesBuilder, with operation: Operation) {
-        routesBuilder.on(operation.httpMethod, [], use: self.handleRequest)
+        routesBuilder.on(Vapor.HTTPMethod(operation), [], use: self.handleRequest)
     }
 
-    func handleRequest(request: Vapor.Request) -> EventLoopFuture<ResponseContainer> {
-        var context = self.contextCreator()
-        let response = context.handle(request: request)
+    func handleRequest(request: Vapor.Request) -> EventLoopFuture<Vapor.Response> {
+        let context = contextCreator()
 
-        var links = ["self": configuration.uriPrefix + endpoint.absolutePath.asPathString()]
-        for relationship in endpoint.relationships {
-            links[relationship.name] = configuration.uriPrefix + relationship.destinationPath.asPathString()
-        }
+        let responseFuture = context.handle(request: request)
 
-        return response.map { encodableAction in
+        return responseFuture.flatMap { (encodableAction: Response<HandledRequest>) in
```

Comment:
I didn't quite follow the ApodiniError PR that closely. Is there additionally work required?
In any case it will be forwarded to Vapor creating some sort of error formatting, but not sure if this is the expected one.

Reply:
As far as there might be some small work that needs to be done to handle the error in the exporters. The exporter must conform to `StandardErrorCompliantExporter`. 

But this is definitely out of scope of the PR here. The WebSocket exporters showcases how a more custom error handling is implemented there 👍

Reply:
Thanks for the clarification.
Yeah, would definitely put that into a separate PR 😅 
Maybe I can squeeze something in, next week. Otherwise it will be an Issue.